### PR TITLE
Admin: Mentor SM fix, Further separation of ahelps and mhelps, Better note timestamps, Shorter log download cooldown, Admins can permaban, Randomized new humans

### PR DESCRIPTION
**Pull request by DominikPanic**

### CODE REVIEW

**Comment by Alterist** [comment]
So,mentors can only SM by right clicking and selecting the Subtle message, they can't use the player panel or chat option. I assume this fixed that?

**Comment by DominikPanic** [comment]
> 
> 
> So,mentors can only SM by right clicking and selecting the Subtle message, they can't use the player panel or chat option. I assume this fixed that?

Yes.

File: code/modules/admin/verbs/randomverbs.dm [hunk 1]
```diff
@@ -48,17 +48,18 @@
...
 		return
 
 	var/msg = input("Message:", text("Subtle PM to [M.key]")) as text
 
-	if (!msg)
+	if(!msg)
 		return
 	if(usr)
-		if (usr.client)
+		if(usr.client)
```
Comment (Rohesie) [inline code/modules/admin/verbs/randomverbs.dm]: This is the same as `if(usr && usr.client && usr.client.holder)` but can be shortened for better performance as `if(usr?.client?.holder)`

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -22,6 +22,110 @@
 		else
 			set_species()
 
+	switch(pick(50;"female", 50;"male"))
+		if("female")
+			gender = FEMALE
+			name = pick(first_names_female) + " " + pick(last_names)
+			real_name = pick(first_names_female) + " " + pick(last_names)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/human/human.dm]:
```suggestion
			real_name = name
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 3]
```diff
@@ -22,6 +22,110 @@
...
+			voice_name = pick(first_names_female) + " " + pick(last_names)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/human/human.dm]:
```suggestion
			voice_name = name
```

Comment (Rohesie) [reply code/modules/mob/living/carbon/human/human.dm]: Else they'll have 3 different names.

File: code/modules/mob/living/carbon/human/human.dm [hunk 4]
```diff
@@ -22,6 +22,110 @@
...
+
+		if("male")
+			gender = MALE
+			name = pick(first_names_male) + " " + pick(last_names)
+			real_name = pick(first_names_male) + " " + pick(last_names)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/human/human.dm]:
```suggestion
			real_name = name
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 5]
```diff
@@ -22,6 +22,110 @@
...
+			voice_name = pick(first_names_male) + " " + pick(last_names)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/human/human.dm]:
```suggestion
			voice_name = name
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 6]
```diff
@@ -22,6 +22,110 @@
...
+			r_hair = 128
+			g_hair = 0
+			b_hair = 0
+			r_facial = 128
+			g_facial = 0
+			b_facial = 0
+
+	h_style = random_hair_style(gender)
+
+	switch(pick(50;"none", 50;"some"))
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/human/human.dm]:
```suggestion
	switch(pick("none", "some"))
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 7]
```diff
@@ -22,6 +22,110 @@
 		else
 			set_species()
 
+	switch(pick(50;"female", 50;"male"))
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/human/human.dm]:
```suggestion
	switch(pick("female", "male"))
```
No need to assign values if they have the same probability.

File: code/modules/mob/mob_transformation_simple.dm [hunk 8]
```diff
@@ -14,26 +14,26 @@
...
 		to_chat(usr, "\red cannot convert into a new_player mob type.")
 		return
 
 	var/mob/M
 	if(isturf(location))
-		M = new new_type( location )
+		M = new new_type(location)
 	else
-		M = new new_type( src.loc )
+		M = new new_type(src.loc)
```
Comment (Rohesie) [inline code/modules/mob/mob_transformation_simple.dm]:
```suggestion
		M = new new_type(loc)
```

File: code/modules/mob/mob_transformation_simple.dm [hunk 9]
```diff
@@ -47,13 +47,35 @@
 		mind.transfer_to(M, TRUE) // second argument to force key move to new mob)
 	else
 		M.key = key
-		if(M.client) M.client.change_view(world.view)
+		if(M.client)
```
Comment (Rohesie) [inline code/modules/mob/mob_transformation_simple.dm]:
```suggestion
```

File: code/modules/mob/mob_transformation_simple.dm [hunk 10]
```diff
@@ -47,13 +47,35 @@
 		mind.transfer_to(M, TRUE) // second argument to force key move to new mob)
 	else
 		M.key = key
-		if(M.client) M.client.change_view(world.view)
+		if(M.client) 
+			M.client.change_view(world.view)
```
Comment (Rohesie) [inline code/modules/mob/mob_transformation_simple.dm]:
```suggestion
		M.client?.change_view(world.view)
```

**Pull request merged**